Update Sonar and PSD based Self-Driving algorithm

Jihoo-J opened the pull request.

Jihoo-J committed:
PSD Algorithm
- using normalized analog value

Jihoo-J committed:
Update Sonar and PSD based Self-Driving algorithm

- Increase default speed from 450 to 550
- Add PSD values to desired speed for PID

yunkon-kim commented on cart/Src/main.c:
```diff
@@ -126,12 +132,10 @@ float Error_R=0;
 float Old_Motor_L;
 float Old_Motor_R;
 
+/************FOR PID*************/
```
More description is necessary, such as PID control for cornering and avoiding obstacles.

yunkon-kim commented on cart/Src/main.c:
```diff
@@ -291,28 +293,54 @@ void SONAR(){
+	PSDR[1]=adcval[4];
+	PSDR[2]=adcval[5];
+
+	/**************PSD NORMALIZATION****************/
```
You need to explain the range of normalization to a comment.

Jihoo-J committed:
Update Sonar and PSD control mechanism for Self-Driving

- Increase default speed from 450 to 550
- Add PSD values to desired speed for PID control
